File: src/main/java/particle/jobjects/Field.java
package particle.jobjects;

import java.util.ArrayList;

public class Field {

    private ArrayList<Particle> particles;

    Field() {
        particles = new ArrayList<>();
    }

    void add(Particle p) {
        particles.add(p);
    }

    int totalParticles() {
        return particles.size();
    }

    Particle particleByIndex(int i) {
        return particles.get(i);
    }

    void remove(Particle p) {
        particles.remove(p);
    }
}

File: src/main/kotlin/particle/Field.kt
package particle

class Field internal constructor() {

    private val particles: MutableList<Particle> = mutableListOf()

    internal val totalParticles: Int
        get() = particles.size

    internal fun add(p: Particle) {
        particles.add(p)
    }

    internal fun particleByIndex(i: Int): Particle {
        return particles[i]
    }

    internal fun removeAll(toRemove: MutableList<Particle>) {
        particles.removeAll(toRemove)
    }
}